src/redux/modules/Folders/Folders.reducer.ts

```diff
@@ -43,14 +42,34 @@ export const reducers = {
       (folderApplet) => folderApplet.id !== action.payload.id,
     );
   },
+  updateAppletData: (
+    state: FoldersSchema,
+    action: PayloadAction<{ appletId: string; published?: boolean; appletName?: string }>,
+  ): void => {
+    const appletToUpdate = state.flattenFoldersApplets.data.find(
+      (applet) => applet.id === action.payload.appletId,
+    );
+    const updatedApplet = {
+      ...appletToUpdate,
+      published: action.payload.published,
+      name: action.payload.appletName || appletToUpdate?.name,
+      updated: new Date().toLocaleString(),
+    };
+    if (updatedApplet?.id) {
+      state.flattenFoldersApplets.data = [
```

Comment:
Can we just use map to update the flattenFoldersApplets state? I think it would be more readable

Reply:
done